### PR TITLE
Show new syncing tables as greyed out in data reference pane

### LINKED ISSUE #32591: Show new syncing tables as greyed out in data reference pane
While working on https://github.com/metabase/metabase/issues/31082, I found that newly syncing tables were not greyed out in the data reference pane as described in the [original product doc](https://www.notion.so/metabase/Provide-progress-on-initial-database-sync-16af8e79312a45ecb715d8e58801c52b?pvs=4#5d67f984521843c4818e17a37719a111).

Instead, they appear as normal and are clickable, even if the fields do not exist.

Any tables with `initial_sync_status!='complete'` should be grayed out as other places in the app, like in the table browser (see https://github.com/metabase/metabase/pull/31609).

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #32591

### Description

Disables and grays out links to not-fully-synced tables in native query editor's right-hand-side sidebar.

Design doc: https://www.notion.so/metabase/Provide-progress-on-initial-database-sync-16af8e79312a45ecb715d8e58801c52b#5d67f984521843c4818e17a37719a111

#### Before

image

#### After

image

### How to verify

1. Simulate long sync with this patch (as shown in [this video](https://www.loom.com/share/e3e9b5a2c2a441aaba4c94ca4f01b776) from [this PR](https://github.com/metabase/metabase/pull/31609)) - _make sure to rebuild_:
   ```diff
   diff --git a/src/metabase/sync/sync_metadata/fields.clj b/src/metabase/sync/sync_metadata/fields.clj
   index 58147f61eb..55afecc5b0 100644
   --- a/src/metabase/sync/sync_metadata/fields.clj
   +++ b/src/metabase/sync/sync_metadata/fields.clj
   @@ -80,6 +80,7 @@
                                    :total-fields   su/IntGreaterThanOrEqualToZero})
      "Sync the Fields in the Metabase application database for all the Tables in a `database`."
      [database :- i/DatabaseInstance]
   +  (Thread/sleep 30000)
      (->> database
         sync-util/db->sync-tables
         (map (partial sync-fields-for-table! database))
   ```
2. Spin up the QA dbs:
    ```Shell
    yarn test-qa-dbs:up
    ````
3. Go to http://localhost:3000/admin/databases/create and connect Postgres QA DB ([credentials](https://github.com/metabase/metabase/blob/d6473cdce794254293e4be2cc4f17aef527ccf05/e2e/support/cypress_data.js#L164-L170)).
4. Now, within 30 seconds, click: "**+ New**" > "**SQL query**" > "**Learn more about your data**" > "**{name-of-the database-connected-in-the-previous-step}**"
5. You should see a list of 4 tables in that database.

Expected result: the tables are grayed out and non-interactive.

### Demo

https://github.com/metabase/metabase/assets/6830683/9e3ec84d-1e87-4e5e-b02e-a9acd590719e

### CODE REVIEW

File: frontend/src/metabase/entities/tables.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ import Metrics from "metabase/entities/metrics";
 import Segments from "metabase/entities/segments";
 import Questions from "metabase/entities/questions";
 
-import { GET, PUT } from "metabase/lib/api";
+import { PUT } from "metabase/lib/api";
```
Comment (kamilmielnik) [inline frontend/src/metabase/entities/tables.js:24]: Changes in this file are unrelated to the PR.
I reused existing `MetabaseApi` definitions of endpoints.
There should be no difference in how anything works here.

**Review by ranquild — APPROVED** [review]
Could we check that some items are disabled instead of userEvent.click throwing? Without any context it's hard to understand that one means another.

There is also a dedicated section of the app called "data reference" available at `/reference`. Here items aren't disabled. Maybe worth creating a follow-up PR

**Review by calherries — APPROVED** [review]
LGTM

**Comment by kamilmielnik** [comment]
> Could we check that some items are disabled instead of userEvent.click throwing? Without any context it's hard to understand that one means another.

I updated the code to better communicate the intention and reason for this: 42c690a738fb531e2e6931174e69519d190bcb71

> There is also a dedicated section of the app called "data reference" available at `/reference`. Here items aren't disabled. Maybe worth creating a follow-up PR

I created a separate issue for it: https://github.com/metabase/metabase/issues/33152
